### PR TITLE
Adding custom dockerfile for when we need to build Dyninst from GitHub

### PR DESCRIPTION
**Pull request by vsoch**
This is just putting the Dyninst dockerfile alongside the code for when we need it.

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

File: Dockerfile.dyninst [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Build stage with Spack pre-installed and ready to be used
+# docker build -f Dockerfile.dyninst -t ghcr.io/autamus/dyninst:05-25-master .
+FROM spack/ubuntu-bionic:latest as builder
+
+RUN git clone https://github.com/autamus/registry && \
```
Comment (hainest) [inline Dockerfile.dyninst:5]: I'm assuming autamus keeps a regular build of the HEAD of master for Dyninst?

Comment (vsoch) [reply Dockerfile.dyninst:5]: it does not - it only builds releases. We need this Dockerfile so we can get a build for the base (with your fixes).

File: Dockerfile.dyninst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+RUN apt-get -yqq update && apt-get -yqq upgrade \
+ && apt-get -yqq install python3-boto3 \
+ && rm -rf /var/lib/apt/lists/*
+
+# What we want to install and how we want to install it
+# is specified in a manifest file (spack.yaml)
+RUN mkdir /opt/spack-environment \
+&&  (echo "spack:" \
+&&   echo "  specs:" \
+&&   echo "  - dyninst@master" \
```
Comment (vsoch) [inline Dockerfile.dyninst:19]: @hainest this is what decides to build from the HEAD of dyninst on master.

Comment (hainest) [reply Dockerfile.dyninst:19]: Ah, ok. I didn't notice this part!

**Review by hainest — APPROVED** [review]

**Pull request merged**